src/NodeServices/PythonServices.cs

```diff
@@ -219,6 +219,14 @@ private void LoadPythonEngine(Assembly assembly)
 
                     instanceProp = eType?.GetProperty(PythonEvaluatorSingletonInstance, BindingFlags.NonPublic | BindingFlags.Static);
                     if (instanceProp == null) return;
+
+                    //at this point we have found an assembly that contains a python engine
+                    //and is correctly formed, let's see if we can load its dependencies now
+                    //this will help test that it will not fail to load later outside of a try / catch for example.
+                    foreach (var asm in assembly.GetReferencedAssemblies())
+                    {
+                        Assembly.Load(asm);
+                    }
```

Comment:
I wonder if there are cases when missing assembly references could be considered ok.
Like if there is dead code (but still references a missing binary), should that be considered ok?

Reply:
also if we do this, we will probably need to load them all in the same ALC as the original assemby (when we do isolation)

Reply:
Also I see the new code is in a try catch {} which ignoners all exceptions. How did you get that exception to show in your screen shot ?

Reply:
Would it make sense to load them in a metadataload context ? and discard it...
I am not sure that would add any value...

Reply:
It's not clear what exact behavior that will have but we can try it.

Reply:
@pinzart90 shifted to using an MLC - PTAL.